src/clj/puppetlabs/trapperkeeper/services/watcher/filesystem_watch_core.clj

```diff
@@ -35,27 +35,32 @@
 
 (schema/defn clojurize :- Event
   [event :- WatchEvent
-   watch-path :- Path]
-  {:type (get event-type-mappings (.kind event))
-   :path (when-not (= StandardWatchEventKinds/OVERFLOW (.kind event))
-           (-> watch-path
-               (.resolve (.context event))
-               fs/file))})
-
-;; This is quite similar to the function above but it is more a direct
-;; conversion of the exact data available on a specific WatchEvent instance,
-;; only used for debugging.
-(defn clojurize-for-logging
-  [e]
-  {:context (.context e)
-   :count (.count e)
-   :kind (.kind e)})
-
-(defn pprint-events
-  [events]
-  (->> events
-       (map #(update % :path str))
-       ks/pprint-to-string))
+   watched-path :- Path]
+  (let [kind (get event-type-mappings (.kind event))
+        changed-path (when-not (= :unknown kind)
+                       (.context event))
+        full-path (when-not (= :unknown kind)
+                    (-> watched-path (.resolve changed-path) fs/file))]
+    {:type kind
+     :count (.count event)
+     :watched-path (fs/file watched-path)
+     :changed-path (fs/file changed-path)
+     :full-path full-path}))
+
+(schema/defn format-for-debugging
+  [{:keys [changed-path count type]} :- Event]
+  {:context changed-path
+   :count count
+   :kind type})
+
+(schema/defn format-for-consumers
+  [{:keys [full-path type]} :- Event]
+  {:path full-path
+   :type type})
+
+(schema/defn format-for-users
```

Comment:
The name of this function seems misleading since it is only used in a log statement.

Also, it doesn't seem to have much effect except logging a stringified-file instead of a `File` object.  It doesn't seem usefully-different from `format-for-consumers` - and I wonder if the logging call which uses it could just be deleted/combined with the other one at `trace` immediately after it.